src/compiler/declarationEmitter.ts

```diff
@@ -1352,6 +1360,15 @@ namespace ts {
             // If we are emitting Method/Constructor it isn't moduleElement and hence already determined to be emitting
             // so no need to verify if the declaration is visible
             if (!resolver.isImplementationOfOverload(node)) {
+                // If we are emitting a private method or constructor it might be already emitted with an overloaded signature

```

Comment:
This seems incorrect way of handling it.
You could just check in emitSignatureDeclaration if this is first signature and skip rest for private methods.
Map of name is not needed and if needed you can ask resolver for the question instead.